arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc

```diff
@@ -5,18 +5,29 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+config SOC_NRF52810
```

Comment:
There is one more difference between nRF52810 and other members of the nRF52 family: The nRF52810 does not have instruction cache, which should be enabled on other SoCs.

At the moment it looks that we are not enabling cache at all. IMHO we should fix that asap as without cache performance is significantly degraded. Please fix that in this PR or create new issue.